python/cudf/cudf/tests/test_struct.py

```diff
@@ -332,3 +332,42 @@ def test_nested_struct_from_pandas_empty():
     gdf = cudf.from_pandas(pdf)
 
     assert_eq(pdf, gdf)
+
+
+@pytest.mark.parametrize(
+    "data, idx, expected",
```

Comment:
Same as above, no need to define `expected`. Use `data[idx]`.

Reply:
Here, we would need `expected`. Because in the second test case:
```python
(Pdb) series[idx]
{'f1': <NA>, 'f2': {'a': 'sf21'}}
(Pdb) data[idx]
{'f2': {'a': 'sf21'}}
```

Reply:
~We could probably write:~
```python
all_keys = {col for row in data for col in row.keys()}
expected = {col: data[idx].get(col, cudf.NA) for col in all_keys}
```
~Pro: doesn't rely on manual specification of data, simpler parameterization~
~Con: requires a bit of reading to understand how the expectation value is constructed~

~Either path is fine here.~

edit: this doesn't work for lists of structs, I give up. Manual specification wins!